scripts/migration/migrate.py

```diff
@@ -117,6 +117,110 @@ def add_v1_space_members_to_v2_group(space, group_id, headers):
         )
 
 
+def get_clowder_v1_user_collections(headers, user_v1):
+    endpoint = f"{CLOWDER_V1}/api/collections"
+    response = requests.get(endpoint, headers=headers)
+    return [col for col in response.json() if col["authorId"] == user_v1["id"]]
+
+
+def get_clowder_v1_dataset_collections(headers, user_v1, dataset_id):
+    matching_collections = []
+    endpoint = f"{CLOWDER_V1}/api/collections/allCollections"
+    response = requests.get(endpoint, headers=headers)
+    user_collections = response.json()
+    for collection in user_collections:
+        collection_id = collection["id"]
+        if collection["name"] == "subchild":
+            print("HERE")
+        collection_dataset_endpoint = (
+            f"{CLOWDER_V1}/api/collections/{collection_id}/datasets"
+        )
+        try:
+            dataset_response = requests.get(
+                collection_dataset_endpoint, headers=headers
+            )
+            datasets = dataset_response.json()
+            for ds in datasets:
+                if ds["id"] == dataset_id:
+                    matching_collections.append(collection)
+        except Exception as e:
+            print("Exception", e)
+    return matching_collections
+
+
+def get_clowder_v1_collection(collection_id, headers):
+    endpoint = f"{CLOWDER_V1}/api/collections/{collection_id}"
+    response = requests.get(endpoint, headers=headers)
+    return response.json()
+
+
+def get_clowder_v1_collections(collection_ids, headers):
+    collections = []
+    for collection_id in collection_ids:
+        endpoint = f"{CLOWDER_V1}/api/collections/{collection_id}"
+        response = requests.get(endpoint, headers=headers)
+        collections.append(response.json())
+    return collections
+
+
+def get_clowder_v1_collection_self_and_ancestors(
+    collection_id, self_and_ancestors, headers
+):
+    endpoint = f"{CLOWDER_V1}/api/collections/{collection_id}"
+    response = requests.get(endpoint, headers=headers)
+    self = response.json()
+    if "id" not in self:
+        print("HERE")
```

Comment:
same Is this print needed?

Reply:
will remove, i was using this print in debugging